src/app/COMPONENTS/task/task.component.ts

```diff
@@ -26,16 +27,28 @@ import { MaterialModule } from '../../material.module';
 
 
 export class TaskComponent {
-  tasks$ = this.taskService.availableTasks$;
+  availableTasks$ = new BehaviorSubject<Task[]>([])
 
-  constructor(private taskService: TasksService) { }
+  constructor(private taskService: TasksService) {
+  }
+  ngOnInit() {
+    this.taskService.getAllTasks().subscribe(data => {
+      this.availableTasks$.next(data)
+
+    })
+  }
 
+  deleteTask(taskId: string) {
+    this.taskService.deleteTask(taskId)
+    const updatedTasksArray = this.availableTasks$.getValue().filter((task) => task.id !== taskId);
+    this.availableTasks$.next(updatedTasksArray)
```

Comment:
going back to what I said in the previous comment, now you can see how taxing (memory leaks) the app can become, you have multiple subscriptions in the task service and they're not being unsubscribed from. Instead what would be a more efficient (and standard) approach would be to assign the result deleteTask or updateTask returns from the BE:
this.availableTasks$ = this.taskService.deleteTask(taskId) (BE should mostly do all the background work like deleting from DB etc. so it would return a new list of availableTasks, that you will assign to this.availableTasks$)